### PR TITLE
Cleanup hp algorithm

### PR DESCRIPTION
**Pull request by Erwann9875**
Cleanup hp algorithm using leaked .php

### CODE REVIEW

File: src/NosCore.Algorithm/Constants.cs [hunk 1]
```diff
@@ -9,6 +9,7 @@ internal class Constants
         internal const byte MaxFairyLevel = 80;
         internal const byte MaxJobLevel = 80;
         internal const byte MaxHeroLevel = 60;
-        internal static readonly int ClassCount = Enum.GetNames(typeof(CharacterClassType)).Length;
```
Comment (0Lucifer0) [inline src/NosCore.Algorithm/Constants.cs]: this has nothing to do in that file it's not something global but something particular for a specific implementation

File: src/NosCore.Algorithm/HpService/HpService.cs [hunk 2]
```diff
@@ -11,98 +11,23 @@ namespace NosCore.Algorithm.HpService
...
+                    var hp = (0.5 * Math.Pow(hpx, 2)) + (15.5 * hpx) + 205;
+                    _hpData[(int)classType, i] = (long)hp;
                 }
-
-                _hpData[(byte)CharacterClassType.MartialArtist, i] = fighterHp;
             }
         }
         public long GetHp(CharacterClassType @class, byte level)
         {
-            return _hpData![(byte)@class, level-1];
```
Comment (0Lucifer0) [inline src/NosCore.Algorithm/HpService/HpService.cs]: this make more sense for array to start at 0

File: src/NosCore.Algorithm/HpService/HpService.cs [hunk 3]
```diff
@@ -11,98 +11,23 @@ namespace NosCore.Algorithm.HpService
 {
     public class HpService : IHpService
     {
-        private readonly long[,] _hpData = new long[Constants.ClassCount, Constants.MaxLevel];
+        private readonly long[,] _hpData = new long[Constants.ClassCount, Constants.MaxLevel + 1];
```
Comment (0Lucifer0) [inline src/NosCore.Algorithm/HpService/HpService.cs]: you don't use the 0 value in that array so no real reason to add +1

File: src/NosCore.Algorithm/Constants.cs [hunk 4]
```diff
@@ -1,14 +1,11 @@
...
 {
     internal class Constants
     {
         internal const byte MaxLevel = 99;
         internal const byte MaxFairyLevel = 80;
         internal const byte MaxJobLevel = 80;
-        internal const byte MaxHeroLevel = 60;
-        internal static readonly int ClassCount = Enum.GetNames(typeof(CharacterClassType)).Length;
+        internal const byte MaxHeroLevel = 60;
+        internal static readonly int[][] ClassConstants = new int[][] { new int[] { 0, 0, 0 }, new int[] { 8, 2, 0 }, new int[] { 3, 6, 1 }, new int[] { 0, 2, 8 }, new int[] { 5, 3, 2 } };
```
Comment (erwan-joly) [inline src/NosCore.Algorithm/Constants.cs]: this also doesn't have anything to do here

File: src/NosCore.Algorithm/MpService/MpService.cs [hunk 5]
```diff
@@ -11,7 +11,7 @@ namespace NosCore.Algorithm.MpService
 {
     public class MpService : IMpService
     {
-        private readonly long[,] _mpData = new long[Constants.ClassCount, Constants.MaxLevel];
+        private readonly long[,] _mpData = new long[Enum.GetNames(typeof(CharacterClassType)).Length, Constants.MaxLevel];
```
Comment (erwan-joly) [inline src/NosCore.Algorithm/MpService/MpService.cs:14]: just hardcode 5 if we change to 6 the amount of class we will still need to implement it so there is no real reason to get the number from the enum

**Comment by 0Lucifer0** [comment]
LGTM

**Pull request merged**